lib/src/app_handler.dart

```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+
+// Described an specific application which can open a file
+class AppHandler {
```

Comment:
We should annotate this class as `@immutable` and definitely make all instance fields `final`